### ISSUE TITLE
VSR Model Training Issues

### ISSUE THREAD

**Issue by jeonhuhuhu** [mpc001/auto_avsr#22]
We are learning the VSR model as it is without any modifications.

The code you proceed with during your study is as follows.

python train.py exp_dir=[exp_dir] \
               exp_name=[exp_name] \
               data.modality="video" \
               data.dataset.root_dir=[root_dir] \
               data.dataset.train_file="lrs3_train_transcript_lengths_seg24s.csv" \
               data.dataset.val_file="lrs3_test_transcript_lengths_seg24s.csv" \
               trainer.num_nodes="1" \
               trainer.gpus="5" \
               data.max_frames="1800" \
               optimizer.lr="0.0002" \
               

: However, even after training the code several times, the values of "decoder_acc_step" and "decoder_acc_val" do not change when they exceed Epoch 30. 

This means that the loss value does not drop.

Is there anything else important to set up when training in particular?

Thank you for your response in advance.

**Comment by mpc001** [comment]
Hi @jeonhuhuhu,

The learning rate 0.0002 is only used when training the subset of lrs3, in that case, I think `data.dataset.train_file` should be set to `lrs3_train_transcript_lengths_seg24s_0to100.csv`. It's also worth noting that we've included the [vsr_trlrs3_23h_base.pth](https://drive.google.com/file/d/1OBEHbStKKFG7VDij14RDLN9VYSdE_Bhs/view?usp=sharing) in model zoo for your convenience. Please feel free to use the provided checkpoint If you wish to skip this step.


After that, for fine-tuning the pre-trained model on the complete LRS3 dataset, please pass the path to pretrained_model_path as an argument and set the learning rate to 0.001. I provide the command lines below:

python train.py exp_dir=[exp_dir]
exp_name=[exp_name]
data.modality="video"
data.dataset.root_dir=[root_dir]
data.dataset.train_file="lrs3_train_transcript_lengths_seg24s.csv"
data.dataset.val_file="lrs3_test_transcript_lengths_seg24s.csv"
trainer.num_nodes="1"
trainer.gpus="5"
data.max_frames="1800"
optimizer.lr="0.001" 
pretrained_model_path=[model_path_of_vsr_trlrs3_23h_base]

**Comment by jeonhuhuhu** [comment]
> Hi @jeonhuhuhu,
> 
> The learning rate 0.0002 is only used when training the subset of lrs3, in that case, I think `data.dataset.train_file` should be set to `lrs3_train_transcript_lengths_seg24s_0to100.csv`. It's also worth noting that we've included the [vsr_trlrs3_23h_base.pth](https://drive.google.com/file/d/1OBEHbStKKFG7VDij14RDLN9VYSdE_Bhs/view?usp=sharing) in model zoo for your convenience. Please feel free to use the provided checkpoint If you wish to skip this step.
> 
> After that, for fine-tuning the pre-trained model on the complete LRS3 dataset, please pass the path to pretrained_model_path as an argument and set the learning rate to 0.001. I provide the command lines below:
> 
> python train.py exp_dir=[exp_dir] exp_name=[exp_name] data.modality="video" data.dataset.root_dir=[root_dir] data.dataset.train_file="lrs3_train_transcript_lengths_seg24s.csv" data.dataset.val_file="lrs3_test_transcript_lengths_seg24s.csv" trainer.num_nodes="1" trainer.gpus="5" data.max_frames="1800" optimizer.lr="0.001" pretrained_model_path=[model_path_of_vsr_trlrs3_23h_base]

Thank you for your answer.

We also conducted an experiment by referring to your GitHub questions and experimented as the answer above.

When experiments were conducted using only 100 frames or less of data, using your model as it is resulted in 25% decoder_acc_val.

And, using a pretrained model with 25% accuracy, we are re-training the entire data(LRS3) again and have 27% accuracy(decoder_acc_val) at 17 epoch, so I wonder if the training is going well.

If you can share the results of the process of changing training accuracy when you train?
